README.md
# 正则大全  ![已收录84条](https://img.shields.io/badge/已收录-84条-673ab7.svg) [![版本](https://badgen.net/vs-marketplace/v/russell.any-rule)](https://marketplace.visualstudio.com/items?itemName=russell.any-rule) [![安装量](https://badgen.net/vs-marketplace/i/russell.any-rule)](https://marketplace.visualstudio.com/items?itemName=russell.any-rule) [![下载量](https://badgen.net/vs-marketplace/d/russell.any-rule)](https://marketplace.visualstudio.com/items?itemName=russell.any-rule) ![MIT](https://img.shields.io/badge/license-MIT-F44336.svg) [![CircleCI](https://badgen.net/github/status/any86/any-rule/master/ci/circleci)](https://circleci.com/gh/any86/any-rule)

🦕支持**web** / **vscode** / **idea** / **Alfred Workflow**多平台

## :rocket:web版本
https://any-rule.vercel.app/

## 🍭vscode插件

### 安装
vscode应用商店中搜索"**any-rule**".

### 使用
**方式1:**

1. 按**F1**(mac下fn+F1)打开正则列表.
2. **输入关键词搜索**, 比如"手机".

![GIF](https://user-images.githubusercontent.com/8264787/146724787-888a8666-5d2a-4e5e-b383-aec7c53bd312.gif)

**方式2:**

右键选择"🦕正则大全".

![GIF](https://user-images.githubusercontent.com/8264787/146725447-4d92caed-2dd0-4f11-91d1-8da9cd8ff08b.gif)

**方式3:**

在代码任意位置输入"**@zz**".

![GIF](https://user-images.githubusercontent.com/8264787/146725402-30b34119-b709-4d49-adb2-af8dbb786d3b.gif)

## 👩‍🏫图解正则
<details>
<summary>查看详情</summary>
每次在any-rule中选择正则后会弹出提示, 可点击"🤖图解正则".

![image](https://user-images.githubusercontent.com/8264787/146726200-d97e48c3-df76-4531-8210-b6935fffb997.png)

点击后可以看到正则解析, 方便大家学习.

![image](https://user-images.githubusercontent.com/8264787/146726350-58166bf9-7f7f-4685-86aa-f055d0c0b914.png)

**注意**: 图解直接使用了https://regexper.com, 在此对作者表示敬意和感谢.
</details>

## 社区版本

社区版本非本人维护, 只是其他开发者使用了any-rule整理的正则内容, 如使用出现问题可直接与其开发者联系.

[idea版](https://github.com/zhoriya/idea-rule)

[Alfred Workflow版](https://github.com/cccyb/workflows)

[hyjs: 函数封装版](https://github.com/heiyehk/hyjs/tree/main/packages/utils)

[命令行版本](https://github.com/shenguanjiejie/workwork)

[uTools版本](https://github.com/trentlee0/utools-any-rule)

## :fire:关于PR
欢迎大家PR, 步骤如下:
1. **正则**请在**packages/www/src/RULES.js**中添加.
2. 运行`npm run test:rules`进行测试.
3. 运行`npm run build:md`更新**README.md**.
4. 请务必提交到**develop**分支.

在此感谢大家对**any-rule**做出的贡献! 

## 🍔正则

### 火车车次
```javascript
/^[GCDZTSPKXLY1-9]\d{1,4}$/
```

### 手机机身码(IMEI)
```javascript
/^\d{15,17}$/
```

### 必须带端口号的网址(或ip)
```javascript
/^((ht|f)tps?:\/\/)?[\w-]+(\.[\w-]+)+:\d{1,5}\/?$/
```

### 网址(URL)
```javascript
/^(((ht|f)tps?):\/\/)?([^!@#$%^&*?.\s-]([^!@#$%^&*?.\s]{0,63}[^!@#$%^&*?.\s])?\.)+[a-z]{2,6}\/?/
```

### 统一社会信用代码
```javascript
/^[0-9A-HJ-NPQRTUWXY]{2}\d{6}[0-9A-HJ-NPQRTUWXY]{10}$/
```

### 统一社会信用代码(宽松匹配)(15位/18位/20位数字/字母)
```javascript
/^(([0-9A-Za-z]{15})|([0-9A-Za-z]{18})|([0-9A-Za-z]{20}))$/
```

### 迅雷链接
```javascript
/^thunderx?:\/\/[a-zA-Z\d]+=$/
```

### ed2k链接(宽松匹配)
```javascript
/^ed2k:\/\/\|file\|.+\|\/$/
```

### 磁力链接(宽松匹配)
```javascript
/^magnet:\?xt=urn:btih:[0-9a-fA-F]{40,}.*$/
```

### 子网掩码(不包含 0.0.0.0)
```javascript
/^(254|252|248|240|224|192|128)\.0\.0\.0|255\.(254|252|248|240|224|192|128|0)\.0\.0|255\.255\.(254|252|248|240|224|192|128|0)\.0|255\.255\.255\.(255|254|252|248|240|224|192|128|0)$/
```

### linux"隐藏文件"路径
```javascript
/^\/(?:[^/]+\/)*\.[^/]*/
```

### linux文件夹路径
```javascript
/^\/(?:[^/]+\/)*$/
```

### linux文件路径
```javascript
/^\/(?:[^/]+\/)*[^/]+$/
```

### window"文件夹"路径
```javascript
/^[a-zA-Z]:\\(?:\w+\\?)*$/
```

### window下"文件"路径
```javascript
/^[a-zA-Z]:\\(?:\w+\\)*\w+\.\w+$/
```

### 股票代码(A股)
```javascript
/^(s[hz]|S[HZ])(000[\d]{3}|002[\d]{3}|300[\d]{3}|600[\d]{3}|60[\d]{4})$/
```

### 大于等于0, 小于等于150, 支持小数位出现5, 如145.5, 用于判断考卷分数
```javascript
/^150$|^(?:\d|[1-9]\d|1[0-4]\d)(?:\.5)?$/
```

### html注释
```javascript
/<!--[\s\S]*?-->/g
```

### md5格式(32位)
```javascript
/^[a-fA-F0-9]{32}$/
```

### GUID/UUID
```javascript
/^[a-f\d]{4}(?:[a-f\d]{4}-){4}[a-f\d]{12}$/i
```

### 版本号(version)格式必须为X.Y.Z
```javascript
/^\d+(?:\.\d+){2}$/
```

### 视频(video)链接地址（视频格式可按需增删）
```javascript
/^https?:\/\/(.+\/)+.+(\.(swf|avi|flv|mpg|rm|mov|wav|asf|3gp|mkv|rmvb|mp4))$/i
```

### 图片(image)链接地址（图片格式可按需增删）
```javascript
/^https?:\/\/(.+\/)+.+(\.(gif|png|jpg|jpeg|webp|svg|psd|bmp|tif))$/i
```

### 24小时制时间（HH:mm:ss）
```javascript
/^(?:[01]\d|2[0-3]):[0-5]\d:[0-5]\d$/
```

### 12小时制时间（hh:mm:ss）
```javascript
/^(?:1[0-2]|0?[1-9]):[0-5]\d:[0-5]\d$/
```

### base64格式
```javascript
/^\s*data:(?:[a-z]+\/[a-z0-9-+.]+(?:;[a-z-]+=[a-z0-9-]+)?)?(?:;base64)?,([a-z0-9!$&',()*+;=\-._~:@/?%\s]*?)\s*$/i
```

### 数字/货币金额（支持负数、千分位分隔符）
```javascript
/^-?\d{1,3}(,\d{3})*(\.\d{1,2})?$/
```

### 银行卡号（10到30位, 覆盖对公/私账户, 参考[微信支付](https://pay.weixin.qq.com/wiki/doc/api/xiaowei.php?chapter=22_1)）
```javascript
/^[1-9]\d{9,29}$/
```

### 中文姓名
```javascript
/^(?:[\u4e00-\u9fa5·]{2,16})$/
```

### 英文姓名
```javascript
/(^[a-zA-Z][a-zA-Z\s]{0,20}[a-zA-Z]$)/
```

### 车牌号(新能源)
```javascript
/^[京津沪渝冀豫云辽黑湘皖鲁新苏浙赣鄂桂甘晋蒙陕吉闽贵粤青藏川宁琼使领][A-HJ-NP-Z](?:((\d{5}[A-HJK])|([A-HJK][A-HJ-NP-Z0-9][0-9]{4}))|[A-HJ-NP-Z0-9]{4}[A-HJ-NP-Z0-9挂学警港澳])$/
```

### 车牌号(非新能源)
```javascript
/^[京津沪渝冀豫云辽黑湘皖鲁新苏浙赣鄂桂甘晋蒙陕吉闽贵粤青藏川宁琼使领][A-HJ-NP-Z][A-HJ-NP-Z0-9]{4}[A-HJ-NP-Z0-9挂学警港澳]$/
```

### 车牌号(新能源+非新能源)
```javascript
/^[京津沪渝冀豫云辽黑湘皖鲁新苏浙赣鄂桂甘晋蒙陕吉闽贵粤青藏川宁琼使领][A-HJ-NP-Z][A-HJ-NP-Z0-9]{4,5}[A-HJ-NP-Z0-9挂学警港澳]$/
```

### 手机号(mobile phone)中国(严谨), 根据工信部2019年最新公布的手机号段
```javascript
/^(?:(?:\+|00)86)?1(?:(?:3[\d])|(?:4[5-79])|(?:5[0-35-9])|(?:6[5-7])|(?:7[0-8])|(?:8[\d])|(?:9[1589]))\d{8}$/
```

### 手机号(mobile phone)中国(宽松), 只要是13,14,15,16,17,18,19开头即可
```javascript
/^(?:(?:\+|00)86)?1[3-9]\d{9}$/
```

### 手机号(mobile phone)中国(最宽松), 只要是1开头即可, 如果你的手机号是用来接收短信, 优先建议选择这一条
```javascript
/^(?:(?:\+|00)86)?1\d{10}$/
```

### 日期(宽松)
```javascript
/^\d{1,4}(-)(1[0-2]|0?[1-9])\1(0?[1-9]|[1-2]\d|30|31)$/
```

### 日期(严谨, 支持闰年判断)
```javascript
/^(([0-9]{3}[1-9]|[0-9]{2}[1-9][0-9]{1}|[0-9]{1}[1-9][0-9]{2}|[1-9][0-9]{3})-(((0[13578]|1[02])-(0[1-9]|[12][0-9]|3[01]))|((0[469]|11)-(0[1-9]|[12][0-9]|30))|(02-(0[1-9]|[1][0-9]|2[0-8]))))|((([0-9]{2})(0[48]|[2468][048]|[13579][26])|((0[48]|[2468][048]|[3579][26])00))-02-29)$/
```

### 中国省
```javascript
/^浙江|上海|北京|天津|重庆|黑龙江|吉林|辽宁|内蒙古|河北|新疆|甘肃|青海|陕西|宁夏|河南|山东|山西|安徽|湖北|湖南|江苏|四川|贵州|云南|广西|西藏|江西|广东|福建|台湾|海南|香港|澳门$/
```

### 可以被moment转化成功的时间 YYYYMMDD HH:mm:ss
```javascript
/^\d{4}([/:-\S])(1[0-2]|0?[1-9])\1(0?[1-9]|[1-2]\d|30|31) (?:[01]\d|2[0-3]):[0-5]\d:[0-5]\d$/
```

### email(邮箱)
```javascript
/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
```

### 座机(tel phone)电话(国内),如: 0341-86091234
```javascript
/^(?:(?:\d{3}-)?\d{8}|^(?:\d{4}-)?\d{7,8})(?:-\d+)?$/
```

### 身份证号(1代,15位数字)
```javascript
/^[1-9]\d{7}(?:0\d|10|11|12)(?:0[1-9]|[1-2][\d]|30|31)\d{3}$/
```

### 身份证号(2代,18位数字),最后一位是校验位,可能为数字或字符X
```javascript
/^[1-9]\d{5}(?:18|19|20)\d{2}(?:0[1-9]|10|11|12)(?:0[1-9]|[1-2]\d|30|31)\d{3}[\dXx]$/
```

### 身份证号, 支持1/2代(15位/18位数字)
```javascript
/^\d{6}((((((19|20)\d{2})(0[13-9]|1[012])(0[1-9]|[12]\d|30))|(((19|20)\d{2})(0[13578]|1[02])31)|((19|20)\d{2})02(0[1-9]|1\d|2[0-8])|((((19|20)([13579][26]|[2468][048]|0[48]))|(2000))0229))\d{3})|((((\d{2})(0[13-9]|1[012])(0[1-9]|[12]\d|30))|((\d{2})(0[13578]|1[02])31)|((\d{2})02(0[1-9]|1\d|2[0-8]))|(([13579][26]|[2468][048]|0[048])0229))\d{2}))(\d|X|x)$/
```

### 护照（包含香港、澳门）
```javascript
/(^[EeKkGgDdSsPpHh]\d{8}$)|(^(([Ee][a-fA-F])|([DdSsPp][Ee])|([Kk][Jj])|([Mm][Aa])|(1[45]))\d{7}$)/
```

### 帐号是否合法(字母开头，允许5-16字节，允许字母数字下划线组合
```javascript
/^[a-zA-Z]\w{4,15}$/
```

### 中文/汉字
```javascript
/^(?:[\u3400-\u4DB5\u4E00-\u9FEA\uFA0E\uFA0F\uFA11\uFA13\uFA14\uFA1F\uFA21\uFA23\uFA24\uFA27-\uFA29]|[\uD840-\uD868\uD86A-\uD86C\uD86F-\uD872\uD874-\uD879][\uDC00-\uDFFF]|\uD869[\uDC00-\uDED6\uDF00-\uDFFF]|\uD86D[\uDC00-\uDF34\uDF40-\uDFFF]|\uD86E[\uDC00-\uDC1D\uDC20-\uDFFF]|\uD873[\uDC00-\uDEA1\uDEB0-\uDFFF]|\uD87A[\uDC00-\uDFE0])+$/

CHANGELOG.md
# 更新日志
🚀 [提交问题](https://github.com/any86/any-rule/issues/new)

### 0.3.18(2022-11-07)

- 优化数字/货币金额(支持负数、千分位分隔符)
- 优化小数(支持科学计数)
- 优化缩短"用户名校验"正则长度
- 增加户口薄正则
- 增加军官证正则
- 修复MAC地址的第二位一定是偶数(#264)

### 0.3.13(2022-01-10)
- 修正"网址"正则, 兼容更多符合的网址.
- 修正澳门身份证错误.

### 0.3.12(2021-10-20)
- 修复网址中带有"()"不能通过.
- 新增"整数"/"浮点数"等正则.

### 0.3.11(2021-09-26)
- 修复新版vscode中"@zz"字符失效.
- 优化"身份证15/18"正则.

### 0.3.10(2021-05-28)
- 禁止"日期"可以通过"00"月和"00"日
### 0.3.9(2021-04-20)
- 新增"GUID/UUID"正则.
- 修复ip中的"."没有被转义.


### 0.3.8(2021-01-17)
- 优化一些冗余的正则拼写
- 修复新能源汽车不匹配第三位为字母的情况.
- 座机电话兼容带分机号的情况.

### 0.3.7(2020-11-25)
- 修复"IPv4"不严格, 且增加端口校验.
- 新增"座机电话(区号可选)"

### 0.3.6(2020-08-03)
- 新增"港澳台身份证"

### 0.3.5(2020-07-18)
- 修复"车牌号"可以通过"浙苏H6F681"的问题
- 新增"数字和英文字母组成，并且同时含有数字和英文字母"

### 0.3.4(2020-06-14)
- 修正"车牌号(新能源+非新能源)"未对长度进行限制.
- 修正"身份证"中月份可以匹配00月的bug.
- 新增"匹配连续重复的字符"正则.
- 修正座机支持4位区号，8位电话号.
- 修正, 根据用户需求, "qq@qq"样式的email邮箱地址不再通过验证.

### 0.3.3(2020-04-28)
- 优化增加"身份证中日期的约束".
- 优化"html注释|座机|密码"等正则.
- 新增"mac地址"正则.

### 0.3.2(2020-04-03)
- 修复"linux文件夹"正则不能匹配"/".
- 部分正则标题增加"英文关键字", 方便检索.

### 0.3.1 (2020-03-12)
- 修复"@zz唤醒"和"菜单唤醒"在某些机型下失效的bug.
- 修复"设置后未生效".
- 修改部分提示图标.
- 合并'linux文件'和'linux文件夹'正则.
- 新增'linux隐藏文件'正则.

### 0.3.0 (2020-02-28)
- 修改"zz."触发为"@zz"触发.
- 增加右键唤醒正则菜单.
![](https://user-gold-cdn.xitu.io/2020/2/28/1708764046b38231?w=381&h=413&f=png&s=73047)

### 0.2.0 (2020-02-21)

- 优化"zz."体验, 改用"quickPick"组件以支持"模糊搜索".
![gif](https://user-gold-cdn.xitu.io/2020/2/23/1706df78b18466fd?w=954&h=372&f=gif&s=1732199)
- 增加"设置"功能, 可以在设置中搜索"any-rule".
- 新增"🦕图解正则".
![](https://user-gold-cdn.xitu.io/2020/2/23/1706e32c3a6fb116?w=533&h=95&f=png&s=21563)

![](https://user-gold-cdn.xitu.io/2020/2/23/1706e349b600c28b?w=1151&h=500&f=png&s=45210)

### 0.1.0
- 新增支持在代码任意位置输入"zz."唤醒正则列表.

### 0.0.13
- 给"统一社会信用代码"增加^$标记
- 修复图片/视频连接匹配不严谨
- 增加"机身码IMEI"正则
- 增加"火车车次"正则

### 0.0.12
- 增加"必须带端口号的网址(或ip)"正则
- 修复网址可以匹配"...."的错误

### 0.0.11
- 增加"统一社会信用代码"正则

### 0.0.10
- 迅雷正则增加thunderx规则的支持

### 0.0.9
- 修复"ed2k"正则错误

### 0.0.8
- 优化车牌号正则

### 0.0.7
- 更新"银行卡"的匹配长度为10-30位,参考[微信支付](https://pay.weixin.qq.com/wiki/doc/api/xiaowei.php?chapter=22_1)

### 0.0.6
- 修复"手机号(严禁)正则"错误.

### 0.0.5
- 增加正则"迅雷链接" / "ed2k连接" / "磁力链接" / "子网掩码" / "linux文件(夹)路径" / "window文件(夹)路径"


### 0.0.4
- 优化"大于0, 小于150, 支持小数位出现5"减少没必要的捕获
- 修复"html注释"没有匹配换行符

### 0.0.3

- 解决vscode低版本兼容问题
- 替换主页的git图地址

src/RULES.js
module.exports = [{
        title: 'A股代码',
        rule: /^(s[hz]|S[HZ])(000[\d]{3}|002[\d]{3}|300[\d]{3}|600[\d]{3}|60[\d]{4})$/,
        example: 'sz000858、SZ002136、sz300675、SH600600、sh601155'
    },
    {
        title: '大于0, 小于150, 支持小数位出现5, 如145.5, 用于判断考卷分数',
        rule: /^150$|(^\d|^[1-9]\d|^[1][0-4]\d)(\.5)?$/,
        example: '150, 100.5'
    },
    {
        title: 'html注释',
        rule: /<!--(.*?)-->$/,
        example: '如: <!--<div class="_bubble"></div>-->'
    },
    {
        title: 'md5格式(32位)',
        rule: /^[a-f0-9]{32}$/,
        example: '21fe181c5bfc16306a6828c1f7b762e8',
    },
    {
        title: '版本号格式必须为X.Y.Z',
        rule: /^\d+(\.\d+){2}$/,
        example: '如: 16.3.10'
    },
    {
        title: '视频链接地址（视频格式可按需增删）',
        rule: /^https?:\/\/.*?(swf|avi|flv|mpg|rm|mov|wav|asf|3gp|mkv|rmvb|mp4)$/i,
        example: '如: http://www.abc.com/video/wc.avi'
    },
    {
        title: '图片链接地址（图片格式可按需增删）',
        rule: /^https?:\/\/.*?(gif|png|jpg|jpeg|webp|svg|psd|bmp|tif)$/i,
        example: '如: https://www.abc.com/logo.png'
    },
    {
        title: '24小时制时间（HH:mm:ss）',
        rule: /^((?:[01]\d|2[0-3]):[0-5]\d:[0-5]\d$)/,
        example: '如: 23:34:55'
    },
    {
        title: '12小时制时间（hh:mm:ss）',
        rule: /^(1[0-2]|0?[1-9]):[0-5]\d:[0-5]\d$/,
        example: '如: 11:34:55合法、23:34:55不合法'
    },
    {
        title: 'base64格式',
        rule: /^\s*data:([a-z]+\/[a-z0-9-+.]+(;[a-z-]+=[a-z0-9-]+)?)?(;base64)?,([a-z0-9!$&',()*+;=\-._~:@/?%\s]*?)\s*$/i,
        example: 'data:image/gif;base64,xxxx=='
    },
    {
        title: '数字/货币金额（支持负数、千分位分隔符）',
        rule: /(^[-]?[1-9]\d{0,2}($|(,\d{3})*($|(\.\d{1,2}$))))|((^[0](\.\d{1,2})?)|(^[-][0]\.\d{1,2}))$/,
        example: '如: 100、-0.99、3,234.32、-1,900,235.09'
    },
    {
        title: '数字/货币金额 (只支持正数、不支持校验千分位分隔符)',
        rule: /(^[1-9]([0-9]+)?(\.[0-9]{1,2})?$)|(^(0){1}$)|(^[0-9]\.[0-9]([0-9])?$)/,
        example: '如: 0.99、8.99、666'
    },
    {
        title: '银行卡号（16或19位）',
        rule: /^([1-9]{1})(\d{15}|\d{18})$/,
        example: '如: 6222026006705354217'
    },
    {
        title: '中文姓名',
        rule: /^([\u4e00-\u9fa5·]{2,16})$/,
        example: '如: 葛二蛋、凯文·杜兰特、德克·维尔纳·诺维茨基'
    },
    {
        title: '英文姓名',
        rule: /(^[a-zA-Z]{1}[a-zA-Z\s]{0,20}[a-zA-Z]{1}$)/,

        example: '如: James、Kevin Wayne Durant、Dirk Nowitzki'
    },
    {
        title: '新能源车牌号',
        rule: /[京津沪渝冀豫云辽黑湘皖鲁新苏浙赣鄂桂甘晋蒙陕吉闽贵粤青藏川宁琼使领 A-Z]{1}[A-HJ-NP-Z]{1}(([0-9]{5}[DF])|([DF][A-HJ-NP-Z0-9][0-9]{4}))$/,
        example: '如: 京AD92035(小型)、甘G23459F(大型)',
    },
    {
        title: '非新能源车牌号',
        rule: /^[京津沪渝冀豫云辽黑湘皖鲁新苏浙赣鄂桂甘晋蒙陕吉闽贵粤青藏川宁琼使领 A-Z]{1}[A-HJ-NP-Z]{1}[A-Z0-9]{4}[A-Z0-9挂学警港澳]{1}$/,
        example: '如: 京A00599、黑D23908'
    },
    {
        title: '车牌号(新能源+非新能源)',
        rule: /^([京津沪渝冀豫云辽黑湘皖鲁新苏浙赣鄂桂甘晋蒙陕吉闽贵粤青藏川宁琼使领 A-Z]{1}[A-HJ-NP-Z]{1}(([0-9]{5}[DF])|([DF]([A-HJ-NP-Z0-9])[0-9]{4})))|([京津沪渝冀豫云辽黑湘皖鲁新苏浙赣鄂桂甘晋蒙陕吉闽贵粤青藏川宁琼使领 A-Z]{1}[A-Z]{1}[A-HJ-NP-Z0-9]{4}[A-HJ-NP-Z0-9 挂学警港澳]{1})$/,
        example: '如: 京A12345D、京A00599'
    },
    {
        title: 'URL链接(网址)',
        rule: /^((https?|ftp|file):\/\/)?([\da-z.-]+)\.([a-z.]{2,6})(\/\w\.-]*)*\/?/,
        example: '如: www.qq.com 、file:///C:/Users/KD/index.js'
    },
    {
        title: '中国手机号(严谨), 根据工信部2019年最新公布的手机号段',
        rule: /^((\+|00)86)?1((3[\d])|(4[5,6,7,9])|(5[0-3,5-9])|(6[5-7])|(7[0-8])|(8[\d])|(9[1,8,9]))\d{8}$/,
        example: '如: 18311006933、17888829981'
    },
    {
        title: '中国手机号(宽松), 只要是13,14,15,16,17,18,19开头即可',
        rule: /^((\+|00)86)?1[3-9]\d{9}$/,
        example: '如: 18311006933、17888829981'
    },
    {
        title: '中国手机号(最宽松), 只要是1开头即可, 如果你的手机号是用来接收短信, 优先建议选择这一条',
        rule: /^((\+|00)86)?1\d{10}$/,
        example: '如: 18311006933、17888829981'
    },
    {
        title: '日期, 如: 2000-01-01或2000-1-1',
        rule: /^\d{4}(-)(1[0-2]|0?\d)\1([0-2]\d|\d|30|31)$/,
        example: '如: 1990-12-12、2020-1-1'
    },
    {
        title: 'email地址',
        rule: /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/,
        example: '如: 90203918@qq.com、nbilly@126.com'
    },
    {
        title: '国内座机电话,如: 0341-86091234',
        rule: /\d{3}-\d{8}|\d{4}-\d{7}/,
        example: '如: 0936-4211235'
    },
    {
        title: '一代身份证号(15位数字)',
        rule: /^\d{8}(0\d|10|11|12)([0-2]\d|30|31)\d{3}$/,
        example: '如: 622001790131123'
    },
    {
        title: '二代身份证号(18位数字),最后一位是校验位,可能为数字或字符X',
        rule: /^\d{6}(18|19|20)\d{2}(0\d|10|11|12)([0-2]\d|30|31)\d{3}(\d|X|x)$/,
        example: '如: 62222319991205131x'
    },
    {
        title: '身份证号, 支持1/2代(15位/18位数字)',
        rule: /(^\d{8}(0\d|10|11|12)([0-2]\d|30|31)\d{3}$)|(^\d{6}(18|19|20)\d{2}(0\d|10|11|12)([0-2]\d|30|31)\d{3}(\d|X|x)$)/,
        example: '如: 622223199912051311'
    },
    {
        title: '护照（包含香港、澳门）',
        rule: /(^[EeKkGgDdSsPpHh]\d{8}$)|(^(([Ee][a-fA-F])|([DdSsPp][Ee])|([Kk][Jj])|([Mm][Aa])|(1[45]))\d{7}$)/,
        example: '如: s28233515、141234567、159203084、MA1234567、K25345719'
    },
    {
        title: '帐号是否合法(字母开头，允许5-16字节，允许字母数字下划线组合',
        rule: /^[a-zA-Z][a-zA-Z0-9_]{4,15}$/,
        example: '如: justin、justin1989、justin_666'
    },
    {
        title: '纯中文/汉字',
        // rule: /^[\u4E00-\u9FA5]+$/,
        rule: /^(?:[\u3400-\u4DB5\u4E00-\u9FEA\uFA0E\uFA0F\uFA11\uFA13\uFA14\uFA1F\uFA21\uFA23\uFA24\uFA27-\uFA29]|[\uD840-\uD868\uD86A-\uD86C\uD86F-\uD872\uD874-\uD879][\uDC00-\uDFFF]|\uD869[\uDC00-\uDED6\uDF00-\uDFFF]|\uD86D[\uDC00-\uDF34\uDF40-\uDFFF]|\uD86E[\uDC00-\uDC1D\uDC20-\uDFFF]|\uD873[\uDC00-\uDEA1\uDEB0-\uDFFF]|\uD87A[\uDC00-\uDFE0])+$/,
        example: '如: 正则、前端'
    },
    {
        title: '是否小数',
        rule: /^\d+\.\d+$/,
        example: '如: 0.0、0.09'
    },
    {
        title: '电话(座机)',
        rule: /^0\d{2,3}-\d{7,8}$/,
        example: '如: 0936-4211236'
    },
    {
        title: '纯数字',
        rule: /^\d{1,}$/,
        example: '如: 12345678'
    },
    {
        title: '是否html标签',
        rule: /<(.*)>.*<\/\1>|<(.*) \/>/,
        example: '如: <div> </div>'
    },
    {
        title: '是否qq号格式正确',
        rule: /^[1-9][0-9]{4,10}$/,
        example: '如: 903013545、9020304'
    },
    {
        title: '是否由数字和字母组成',
        rule: /^[A-Za-z0-9]+$/,
        example: '如: james666、haha233hi'
    },
    {
        title: '纯英文字母',
        rule: /^[a-zA-Z]+$/,
        example: '如: russel'
    },
    {
        title: '纯小写英文字母组成',
        rule: /^[a-z]+$/,
        example: '如: russel'
    },
    {
        title: '纯大写英文字母',
        rule: /^[A-Z]+$/,
        example: '如: ABC、KD'
    },
    {
        title: '密码强度正则，最少6位，包括至少1个大写字母，1个小写字母，1个数字，1个特殊字符',
        rule: /^.*(?=.{6,})(?=.*\d)(?=.*[A-Z])(?=.*[a-z])(?=.*[!@#$%^&*? ]).*$/,
        example: '如: Kd@curry666'
    },
    {
        title: '用户名正则，4到16位（字母，数字，下划线，减号）',
        rule: /^[a-zA-Z0-9_-]{4,16}$/,
        example: '如: xiaohua_qq'
    },
    {
        title: 'ip-v4',
        rule: /^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/,
        example: '如: 172.16.0.0、127.0.0.0'
    },
    {
        title: 'ip-v6',
        rule: /^((([0-9A-Fa-f]{1,4}:){7}[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){6}:[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){5}:([0-9A-Fa-f]{1,4}:)?[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){4}:([0-9A-Fa-f]{1,4}:){0,2}[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){3}:([0-9A-Fa-f]{1,4}:){0,3}[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){2}:([0-9A-Fa-f]{1,4}:){0,4}[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){6}((\b((25[0-5])|(1\d{2})|(2[0-4]\d)|(\d{1,2}))\b)\.){3}(\b((25[0-5])|(1\d{2})|(2[0-4]\d)|(\d{1,2}))\b))|(([0-9A-Fa-f]{1,4}:){0,5}:((\b((25[0-5])|(1\d{2})|(2[0-4]\d)|(\d{1,2}))\b)\.){3}(\b((25[0-5])|(1\d{2})|(2[0-4]\d)|(\d{1,2}))\b))|(::([0-9A-Fa-f]{1,4}:){0,5}((\b((25[0-5])|(1\d{2})|(2[0-4]\d)|(\d{1,2}))\b)\.){3}(\b((25[0-5])|(1\d{2})|(2[0-4]\d)|(\d{1,2}))\b))|([0-9A-Fa-f]{1,4}::([0-9A-Fa-f]{1,4}:){0,5}[0-9A-Fa-f]{1,4})|(::([0-9A-Fa-f]{1,4}:){0,6}[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){1,7}:))$/i,
        example: '如: 2031:0000:130f:0000:0000:09c0:876a:130b'
    },
    {
        title: '16进制颜色',
        rule: /^#?([a-fA-F0-9]{6}|[a-fA-F0-9]{3})$/,
        example: '如: #f00 、#F90、#000、 #fe9de8'
    },
    {
        title: '微信号，6至20位，以字母开头，字母，数字，减号，下划线',
        rule: /^[a-zA-Z][-_a-zA-Z0-9]{5,19}$/,
        example: '如: github666、kd_-666'
    },
    {
        title: '中国邮政编码',
        rule: /^(0[1-7]|1[0-356]|2[0-7]|3[0-6]|4[0-7]|5[1-7]|6[1-7]|7[0-5]|8[013-6])\d{4}$/,
        example: '如: 734500、100101'
    },
    {
        title: '只包含中文和数字',
        rule: /^((?:[\u3400-\u4DB5\u4E00-\u9FEA\uFA0E\uFA0F\uFA11\uFA13\uFA14\uFA1F\uFA21\uFA23\uFA24\uFA27-\uFA29]|[\uD840-\uD868\uD86A-\uD86C\uD86F-\uD872\uD874-\uD879][\uDC00-\uDFFF]|\uD869[\uDC00-\uDED6\uDF00-\uDFFF]|\uD86D[\uDC00-\uDF34\uDF40-\uDFFF]|\uD86E[\uDC00-\uDC1D\uDC20-\uDFFF]|\uD873[\uDC00-\uDEA1\uDEB0-\uDFFF]|\uD87A[\uDC00-\uDFE0])|(\d))+$/,
        example: '如: 哈哈哈、你好6啊'
    },
    {
        title: '不能包含字母',
        rule: /^[^A-Za-z]*$/,
        example: '如: 你好6啊、@¥()！'
    }
];